### DISCUSSION TITLE
Is --host realy needed?

### DISCUSSION THREAD

**Discussion by bessarabov** [umputun/spot#17] [General]
I'm not sure that I understand why there is a --host option.

To run the `default` task on some host that is not present in spot.yml file I can write (the output of the spot will be written in green color):

```
 spot --host 192.168.1.123:122
 ```
 
 But I can also use the `--target` for that (the output of the spot will be written in blue color):
 
 ```
 spot --target 192.168.1.123:122
 ```
 
 I don't understand what can be done with `--host` that can't be done with `--target`. 

 Maybe `--host` can be removed from the list of spot's options?

**Comment by umputun** [comment]
It is redundant but only in case a single host passed. With `--target` user can quickly set a single destination host but with --host which can be set multiple times (I think, I mentioned it in docs) user can pass multiple hosts, i.e. `--host=example.com --host=h2.example.com`

**Comment by umputun** [comment]
however, the question of why it got a different color is interesting. Generally, color is picked with `CRC[host] % len(colors)`, and I don't see why the way how it passed will change the selection. Let me know if you want to dig it deeper, as it can be sign of some bug

**Comment by bessarabov** [comment]
> It is redundant but only in case a single host passed.

Ok. Now I see that the `--target` can be specified only once, but it allowed more that one `--host`.

```
type options struct {
    // ...
    TargetName   string `short:"d" long:"target" description:"target name" default:"default"`
    // ...
    TargetHosts   []string `short:"h" long:"host" description:"destination host"`
```

Maybe it should be also allowed to specify multiple --target at once? For example, I can imagine a situation when I want to deploy the project to the staging and production in one go:

    spot --target=staging --target=prod

> however, the question of why it got a different color is interesting

Ops. I was using different hosts, this explains the diferent colors.

**Reply by umputun** [reply]
> Maybe it should be also allowed to specify multiple --target at once? For example, I can imagine a situation when I want to deploy the project to the staging and production in one go:

Good idea, agree

**Comment by umputun** [comment]
see #19, should resolve it

**Comment by bessarabov** [comment]
Yes, thank you, this is exactly what I wanted. I'm closing this discussion.
